diff --git content/public/common/common_param_traits_macros.h content/public/common/common_param_traits_macros.h
index 3795fb7dffbe..4370a16fe25c 100644
--- content/public/common/common_param_traits_macros.h
+++ content/public/common/common_param_traits_macros.h
@@ -200,6 +200,7 @@ IPC_STRUCT_TRAITS_BEGIN(content::WebPreferences)
   IPC_STRUCT_TRAITS_MEMBER(save_previous_document_resources)
   IPC_STRUCT_TRAITS_MEMBER(text_autosizing_enabled)
   IPC_STRUCT_TRAITS_MEMBER(double_tap_to_zoom_enabled)
+  IPC_STRUCT_TRAITS_MEMBER(caret_width)
 #if defined(OS_ANDROID)
   IPC_STRUCT_TRAITS_MEMBER(font_scale_factor)
   IPC_STRUCT_TRAITS_MEMBER(device_scale_adjustment)
diff --git content/public/common/web_preferences.h content/public/common/web_preferences.h
index b7da33f1b9a3..8234459aff68 100644
--- content/public/common/web_preferences.h
+++ content/public/common/web_preferences.h
@@ -229,6 +229,8 @@ struct CONTENT_EXPORT WebPreferences {
 
   bool double_tap_to_zoom_enabled;
 
+  int caret_width;
+
 #if defined(OS_ANDROID)
   float font_scale_factor;
   float device_scale_adjustment;
diff --git content/renderer/render_view_impl.cc content/renderer/render_view_impl.cc
index e2e0e58726b8..6bccc551dfa6 100644
--- content/renderer/render_view_impl.cc
+++ content/renderer/render_view_impl.cc
@@ -800,6 +800,8 @@ void RenderView::ApplyWebPreferences(const WebPreferences& prefs,
   settings->SetTextAutosizingEnabled(prefs.text_autosizing_enabled);
   settings->SetDoubleTapToZoomEnabled(prefs.double_tap_to_zoom_enabled);
 
+  settings->SetCaretWidth(prefs.caret_width);
+
 #if defined(OS_ANDROID)
   settings->SetAllowCustomScrollbarInMainFrame(false);
   settings->SetAccessibilityFontScaleFactor(prefs.font_scale_factor);
diff --git third_party/blink/public/web/web_settings.h third_party/blink/public/web/web_settings.h
index e39112b02561..86e1d6dde9fd 100644
--- third_party/blink/public/web/web_settings.h
+++ third_party/blink/public/web/web_settings.h
@@ -150,6 +150,7 @@ class WebSettings {
   virtual void SetAntialiasedClips2dCanvasEnabled(bool) = 0;
   virtual void SetAutoplayPolicy(AutoplayPolicy) = 0;
   virtual void SetAutoZoomFocusedNodeToLegibleScale(bool) = 0;
+  virtual void SetCaretWidth(int) = 0;
   virtual void SetClobberUserAgentInitialScaleQuirk(bool) = 0;
   virtual void SetCookieEnabled(bool) = 0;
   virtual void SetNavigateOnDragDrop(bool) = 0;
diff --git third_party/blink/renderer/core/exported/web_settings_impl.cc third_party/blink/renderer/core/exported/web_settings_impl.cc
index 64085ad40a20..84cb8689bc08 100644
--- third_party/blink/renderer/core/exported/web_settings_impl.cc
+++ third_party/blink/renderer/core/exported/web_settings_impl.cc
@@ -152,6 +152,10 @@ void WebSettingsImpl::SetAutoZoomFocusedNodeToLegibleScale(
       auto_zoom_focused_node_to_legible_scale;
 }
 
+void WebSettingsImpl::SetCaretWidth(int width) {
+  settings_->SetCaretWidth(width);
+}
+
 void WebSettingsImpl::SetTextAutosizingEnabled(bool enabled) {
   dev_tools_emulator_->SetTextAutosizingEnabled(enabled);
 }
diff --git third_party/blink/renderer/core/exported/web_settings_impl.h third_party/blink/renderer/core/exported/web_settings_impl.h
index 7ba728bf622f..2a9a9944e67f 100644
--- third_party/blink/renderer/core/exported/web_settings_impl.h
+++ third_party/blink/renderer/core/exported/web_settings_impl.h
@@ -65,6 +65,7 @@ class CORE_EXPORT WebSettingsImpl final : public WebSettings {
   void SetAntialiased2dCanvasEnabled(bool) override;
   void SetAntialiasedClips2dCanvasEnabled(bool) override;
   void SetAutoZoomFocusedNodeToLegibleScale(bool) override;
+  void SetCaretWidth(int) override;
   void SetClobberUserAgentInitialScaleQuirk(bool) override;
   void SetCookieEnabled(bool) override;
   void SetNavigateOnDragDrop(bool) override;
diff --git third_party/blink/renderer/core/frame/local_frame_view.cc third_party/blink/renderer/core/frame/local_frame_view.cc
index 39ccc46f0378..44df408040ea 100644
--- third_party/blink/renderer/core/frame/local_frame_view.cc
+++ third_party/blink/renderer/core/frame/local_frame_view.cc
@@ -4417,8 +4417,11 @@ void LocalFrameView::ApplyTransformForTopFrameSpace(
 }
 
 LayoutUnit LocalFrameView::CaretWidth() const {
+  int width = 1;
+  if (frame_->GetSettings())
+       width =  GetPage()->GetSettings().GetCaretWidth();
   return LayoutUnit(
-      std::max<float>(1.0, GetChromeClient()->WindowToViewportScalar(1)));
+      std::max<float>(1.0, GetChromeClient()->WindowToViewportScalar(width)));
 }
 
 UkmTimeAggregator& LocalFrameView::EnsureUkmTimeAggregator() {
diff --git third_party/blink/renderer/core/frame/settings.json5 third_party/blink/renderer/core/frame/settings.json5
index 113d998ce550..7a98e0912ab9 100644
--- third_party/blink/renderer/core/frame/settings.json5
+++ third_party/blink/renderer/core/frame/settings.json5
@@ -1010,5 +1010,10 @@
       initial: 800,
       type: "int",
     },
+    {
+      name: "caretWidth",
+      initial: 1,
+      type: "int",
+    },
   ],
 }
