diff --git third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
index 009f3563df63b..40641ca04203a 100644
--- third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
+++ third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc
@@ -161,6 +161,7 @@ bool StructTraits<blink::mojom::WebPreferencesDataView,
   out->double_tap_to_zoom_enabled = data.double_tap_to_zoom_enabled();
   out->fullscreen_supported = data.fullscreen_supported();
   out->text_autosizing_enabled = data.text_autosizing_enabled();
+  out->caret_width = data.caret_width();
 #if BUILDFLAG(IS_ANDROID)
   out->font_scale_factor = data.font_scale_factor();
   out->font_weight_adjustment = data.font_weight_adjustment();
diff --git third_party/blink/public/common/web_preferences/web_preferences.h third_party/blink/public/common/web_preferences/web_preferences.h
index 85f312a9ea1ec..e25920a6209c3 100644
--- third_party/blink/public/common/web_preferences/web_preferences.h
+++ third_party/blink/public/common/web_preferences/web_preferences.h
@@ -244,6 +244,8 @@ struct BLINK_COMMON_EXPORT WebPreferences {
 
   bool text_autosizing_enabled = BUILDFLAG(IS_ANDROID);
 
+  float caret_width = 1;
+
   // Representation of the Web App Manifest scope if any.
   GURL web_app_scope;
 
diff --git third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h
index 1173ace39b625..4015ec8155f49 100644
--- third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h
+++ third_party/blink/public/common/web_preferences/web_preferences_mojom_traits.h
@@ -533,6 +533,10 @@ struct BLINK_COMMON_EXPORT StructTraits<blink::mojom::WebPreferencesDataView,
     return r.text_autosizing_enabled;
   }
 
+  static int caret_width(const blink::web_pref::WebPreferences& r) {
+    return r.caret_width;
+  }
+
   static const GURL& web_app_scope(const ::blink::web_pref::WebPreferences& r) {
     return r.web_app_scope;
   }
diff --git third_party/blink/public/mojom/webpreferences/web_preferences.mojom third_party/blink/public/mojom/webpreferences/web_preferences.mojom
index ffd46eb579c7c..924f37a8227f3 100644
--- third_party/blink/public/mojom/webpreferences/web_preferences.mojom
+++ third_party/blink/public/mojom/webpreferences/web_preferences.mojom
@@ -272,6 +272,8 @@ struct WebPreferences {
 
   bool text_autosizing_enabled;
 
+  float caret_width;
+
   // Representation of the Web App Manifest scope if any.
   url.mojom.Url web_app_scope;
 
diff --git third_party/blink/public/web/web_settings.h third_party/blink/public/web/web_settings.h
index a53b4901dde0d..4dcc1717430d7 100644
--- third_party/blink/public/web/web_settings.h
+++ third_party/blink/public/web/web_settings.h
@@ -112,6 +112,7 @@ class WebSettings {
       bool) = 0;
   virtual void SetAutoZoomFocusedEditableToLegibleScale(bool) = 0;
   virtual void SetCaretBrowsingEnabled(bool) = 0;
+  virtual void SetCaretWidth(float) = 0;
   virtual void SetClobberUserAgentInitialScaleQuirk(bool) = 0;
   virtual void SetCookieEnabled(bool) = 0;
   virtual void SetCursiveFontFamily(const WebString&,
diff --git third_party/blink/renderer/core/exported/web_settings_impl.cc third_party/blink/renderer/core/exported/web_settings_impl.cc
index 4a29a2200eaab..be6f169e702e1 100644
--- third_party/blink/renderer/core/exported/web_settings_impl.cc
+++ third_party/blink/renderer/core/exported/web_settings_impl.cc
@@ -648,6 +648,10 @@ void WebSettingsImpl::SetCaretBrowsingEnabled(bool enabled) {
   settings_->SetCaretBrowsingEnabled(enabled);
 }
 
+void WebSettingsImpl::SetCaretWidth(float caret_width) {
+  settings_->SetCaretWidth(caret_width);
+}
+
 void WebSettingsImpl::SetCookieEnabled(bool enabled) {
   dev_tools_emulator_->SetCookieEnabled(enabled);
 }
diff --git third_party/blink/renderer/core/exported/web_settings_impl.h third_party/blink/renderer/core/exported/web_settings_impl.h
index 5e8d2bfbccd06..d5f3c0c8ea25b 100644
--- third_party/blink/renderer/core/exported/web_settings_impl.h
+++ third_party/blink/renderer/core/exported/web_settings_impl.h
@@ -71,6 +71,7 @@ class CORE_EXPORT WebSettingsImpl final : public WebSettings {
   void SetClobberUserAgentInitialScaleQuirk(bool) override;
   void SetCookieEnabled(bool) override;
   void SetCaretBrowsingEnabled(bool) override;
+  void SetCaretWidth(float) override;
   void SetCursiveFontFamily(const WebString&,
                             UScriptCode = USCRIPT_COMMON) override;
   void SetDNSPrefetchingEnabled(bool) override;
diff --git third_party/blink/renderer/core/exported/web_view_impl.cc third_party/blink/renderer/core/exported/web_view_impl.cc
index f92d9b70fbd32..41279f2ae3881 100644
--- third_party/blink/renderer/core/exported/web_view_impl.cc
+++ third_party/blink/renderer/core/exported/web_view_impl.cc
@@ -1745,6 +1745,7 @@ void WebView::ApplyWebPreferences(const web_pref::WebPreferences& prefs,
   settings->SetFullscreenSupported(prefs.fullscreen_supported);
   settings->SetTextAutosizingEnabled(prefs.text_autosizing_enabled);
   settings->SetDoubleTapToZoomEnabled(prefs.double_tap_to_zoom_enabled);
+  settings->SetCaretWidth(prefs.caret_width);
   blink::WebNetworkStateNotifier::SetNetworkQualityWebHoldback(
       static_cast<blink::WebEffectiveConnectionType>(
           prefs.network_quality_estimator_web_holdback));
diff --git third_party/blink/renderer/core/frame/local_frame_view.cc third_party/blink/renderer/core/frame/local_frame_view.cc
index 2bb55d246950d..256cea3e593da 100644
--- third_party/blink/renderer/core/frame/local_frame_view.cc
+++ third_party/blink/renderer/core/frame/local_frame_view.cc
@@ -4701,8 +4701,11 @@ void LocalFrameView::MapLocalToRemoteMainFrame(
 }
 
 LayoutUnit LocalFrameView::CaretWidth() const {
+  float width = 1.0f;
+  if (frame_->GetSettings())
+    width = frame_->GetSettings()->GetCaretWidth();
   return LayoutUnit(std::max<float>(
-      1.0f, GetChromeClient()->WindowToViewportScalar(&GetFrame(), 1.0f)));
+      1.0f, GetChromeClient()->WindowToViewportScalar(&GetFrame(), width)));
 }
 
 void LocalFrameView::RegisterTapEvent(Element* target) {
diff --git third_party/blink/renderer/core/frame/settings.json5 third_party/blink/renderer/core/frame/settings.json5
index f4cdee12ea435..f44d25b3250d0 100644
--- third_party/blink/renderer/core/frame/settings.json5
+++ third_party/blink/renderer/core/frame/settings.json5
@@ -500,6 +500,12 @@
       invalidate: ["Style"],
     },
 
+    {
+      name: "caretWidth",
+      initial: 1.0,
+      type: "float",
+    },
+
     // Font scale factor for accessibility, applied as part of text autosizing.
     {
       name: "accessibilityFontScaleFactor",
