diff --git third_party/blink/renderer/core/editing/commands/apply_style_command.cc third_party/blink/renderer/core/editing/commands/apply_style_command.cc
index 2cd0db8623c8cbdea487e089da6336b19dad86f1..28c87e19d9281c0bc4710f0690453088131ef2c2 100644
--- third_party/blink/renderer/core/editing/commands/apply_style_command.cc
+++ third_party/blink/renderer/core/editing/commands/apply_style_command.cc
@@ -1183,9 +1183,10 @@ bool ApplyStyleCommand::RemoveInlineStyleFromElement(
   if (IsStyledInlineElementToRemove(element)) {
     if (mode == kRemoveNone)
       return true;
-    if (extracted_style)
+    // do not apply the style to children if we want to remove it
+    /*if (extracted_style)
       extracted_style->MergeInlineStyleOfElement(element,
-                                                 EditingStyle::kOverrideValues);
+                                                 EditingStyle::kOverrideValues);*/
     RemoveNodePreservingChildren(element, editing_state);
     if (editing_state->IsAborted())
       return false;
