diff --git content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm
index 75de452a5f78705a544402fb3a639344665d7ff4..b64d1a29b27b054c8aead9b82e70eda170f90ec0 100644
--- content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm
+++ content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm
@@ -181,6 +181,13 @@ void ExtractUnderlines(NSAttributedString* string,
 
 // RenderWidgetHostViewCocoa ---------------------------------------------------
 
+@interface NSWindow (CefCustomMethods)
+- (int)acceptsFirstMouse;
+@end
+
+constexpr int kStateEnabled = 1;
+constexpr int kStateDisabled = 2;
+
 // Private methods:
 @interface RenderWidgetHostViewCocoa ()
 
@@ -800,6 +807,15 @@ - (AcceptMouseEvents)acceptsMouseEventsOption {
 }
 
 - (BOOL)acceptsFirstMouse:(NSEvent*)theEvent {
+  if ([self.window respondsToSelector:@selector(acceptsFirstMouse)]) {
+    const auto mode = [self.window acceptsFirstMouse];
+    if (mode == kStateEnabled) {
+      return YES;
+    } else if (mode == kStateDisabled) {
+      return NO;
+    }
+  }
+
   // Enable "click-through" if mouse clicks are accepted in inactive windows.
   return
       [self acceptsMouseEventsOption] > AcceptMouseEvents::kWhenInActiveWindow;
