diff --git components/printing/renderer/print_render_frame_helper.cc components/printing/renderer/print_render_frame_helper.cc
index 9119538bae9f..103d7826e335 100644
--- components/printing/renderer/print_render_frame_helper.cc
+++ components/printing/renderer/print_render_frame_helper.cc
@@ -1642,7 +1642,7 @@ void PrintRenderFrameHelper::PrintNode(const blink::WebNode& node) {
 }
 
 void PrintRenderFrameHelper::Print(blink::WebLocalFrame* frame,
-                                   const blink::WebNode& node,
+	const blink::WebNode& node,
 	PrintRequestType print_request_type){
 	Print(frame, node, print_request_type, L"", printing::PageRanges());
 }
@@ -1679,8 +1679,12 @@ void PrintRenderFrameHelper::Print(blink::WebLocalFrame* frame,
 
     PrintMsg_PrintPages_Params print_settings;
     auto self = weak_ptr_factory_.GetWeakPtr();
-    GetPrintSettingsFromUser(frame_ref.GetFrame(), node, expected_page_count,
+	if (printer.empty())
+		GetPrintSettingsFromUser(frame_ref.GetFrame(), node, expected_page_count,
                              print_request_type, &print_settings);
+	else
+		GetPrintSettingsWithPrinterAndPages(frame_ref.GetFrame(), node, expected_page_count,
+			                 printer, pages, print_request_type, &print_settings);
     // Check if |this| is still valid.
     if (!self)
       return;
@@ -2037,10 +2041,13 @@ void PrintRenderFrameHelper::GetPrintSettingsWithPrinterAndPages(
     blink::WebLocalFrame* frame,
     const blink::WebNode& node,
     int expected_pages_count,
-    bool is_scripted,
     const base::string16& printer_name,
     const printing::PageRanges& pages,
+	PrintRequestType print_request_type,
     PrintMsg_PrintPages_Params* print_settings) {
+	bool is_scripted = print_request_type == PrintRequestType::kScripted;
+	DCHECK(is_scripted || print_request_type == PrintRequestType::kRegular);
+
   PrintHostMsg_ScriptedPrint_Params params;
   params.cookie = print_pages_params_->params.document_cookie;
   params.has_selection = frame->HasSelection();
@@ -2049,7 +2056,6 @@ void PrintRenderFrameHelper::GetPrintSettingsWithPrinterAndPages(
   if (IsPrintingNodeOrPdfFrame(frame, node))
     margin_type = GetMarginsForPdf(frame, node, print_pages_params_->params);
   params.margin_type = margin_type;
-  params.is_scripted = is_scripted;
   params.is_modifiable = !IsPrintingNodeOrPdfFrame(frame, node);
   params.pages = pages;
   params.forced_printer = printer_name;
diff --git components/printing/renderer/print_render_frame_helper.h components/printing/renderer/print_render_frame_helper.h
index 510151f113c6..7e2793c8db44 100644
--- components/printing/renderer/print_render_frame_helper.h
+++ components/printing/renderer/print_render_frame_helper.h
@@ -251,16 +251,6 @@ class PrintRenderFrameHelper
   // Notification when printing is done - signal tear-down/free resources.
   void DidFinishPrinting(PrintingResult result);
 
-  // Get final print settings with forced printer and pages to print.
-  // WARNING: |this| may be gone after this method returns.
-  void GetPrintSettingsWithPrinterAndPages(blink::WebLocalFrame* frame,
-                                const blink::WebNode& node,
-                                int expected_pages_count,
-                                bool is_scripted,
-                                const base::string16& printer_name,
-                                const printing::PageRanges& pages,
-                                PrintMsg_PrintPages_Params* print_settings);
-
   // Print Settings -----------------------------------------------------------
 
   // Initialize print page settings with default settings.
@@ -290,6 +280,16 @@ class PrintRenderFrameHelper
                                 int expected_pages_count,
                                 PrintRequestType print_request_type,
                                 PrintMsg_PrintPages_Params* print_settings);
+  
+  // Get final print settings with forced printer and pages to print.
+  // WARNING: |this| may be gone after this method returns.
+  void GetPrintSettingsWithPrinterAndPages(blink::WebLocalFrame* frame,
+                                const blink::WebNode& node,
+                                int expected_pages_count,
+                                const base::string16& printer_name,
+                                const printing::PageRanges& pages,
+	                            PrintRequestType print_request_type,
+                                PrintMsg_PrintPages_Params* print_settings);
 
   // Page Printing / Rendering ------------------------------------------------
 
